test/unit-test/test_derivation.ml

```diff
@@ -241,55 +235,43 @@ let%expect_test "derivation" =
     Context.empty
     (Letrec
       ([
-        ("f", Abs (["x"], App (Var "g", [Var "x"])));
-        ("g", Abs (["x"], App (Var "f", [Var "x"])))
+        ("f", Abs (["X"], App (Var "g", [Var "X"])));
+        ("g", Abs (["X"], App (Var "f", [Var "X"])))
       ], App (Var "f", [Constant (Number 42)])));
   [%expect {|
     (Ok (
-      (TyVar o)
+      (TyVar m)
       (Conj (
         (Conj (
-          (Eq (TyVar a) (TyFun ((TyVar m)) (TyVar n)))
+          (Eq (TyVar a) (TyFun ((TyVar k)) (TyVar l)))
           (Subtype
-            (TyVar o)
-            (TyVar n))
+            (TyVar m)
+            (TyVar l))
           Empty
-          (Subtype (TySingleton (Number 42)) (TyVar m))
+          (Subtype (TySingleton (Number 42)) (TyVar k))
           Empty))
-        (Eq
-          (TyVar a)
-          (TyVar g))
-        (Eq
-          (TyVar g)
-          (TyConstraint
-            (TyFun ((TyVar c)) (TyVar f))
-            (Conj (
-              (Eq (TyVar b) (TyFun ((TyVar d)) (TyVar e)))
-              (Subtype
-                (TyVar f)
-                (TyVar e))
-              Empty
-              (Subtype
-                (TyVar c)
-                (TyVar d))
-              Empty))))
-        (Eq
-          (TyVar b)
-          (TyVar l))
-        (Eq
-          (TyVar l)
-          (TyConstraint
-            (TyFun ((TyVar h)) (TyVar k))
-            (Conj (
-              (Eq (TyVar a) (TyFun ((TyVar i)) (TyVar j)))
-              (Subtype
-                (TyVar k)
-                (TyVar j))
-              Empty
-              (Subtype
-                (TyVar h)
-                (TyVar i))
-              Empty)))))))) |}];
+        (Eq (TyVar a) (TyFun ((TyVar c)) (TyVar f)))
+        (Conj (
+          (Eq (TyVar b) (TyFun ((TyVar d)) (TyVar e)))
+          (Subtype
+            (TyVar f)
+            (TyVar e))
+          Empty
+          (Subtype
+            (TyVar c)
+            (TyVar d))
+          Empty))
+        (Eq (TyVar b) (TyFun ((TyVar g)) (TyVar j)))
+        (Conj (
+          (Eq (TyVar a) (TyFun ((TyVar h)) (TyVar i)))
+          (Subtype
+            (TyVar j)
+            (TyVar i))
+          Empty
+          (Subtype
+            (TyVar g)
+            (TyVar h))
+          Empty)))))) |}];
```

Comment:
I checked this derivation with paper.
![derivation tree](https://pbs.twimg.com/media/DuDDAk8VYAAMfrH.jpg:large)